receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -319,3 +339,47 @@ func (mf *metricFamily) appendMetric(metrics pmetric.MetricSlice) {
 
 	metric.MoveTo(metrics.AppendEmpty())
 }
+
+func (mf *metricFamily) addExemplar(l labels.Labels, e exemplar.Exemplar) {
+	gk := mf.getGroupKey(l)
+	es, ok := mf.exemplars[gk]
+	if !ok {
+		es = pmetric.NewExemplarSlice()
+	}
+	addExemplarToSlice(e, es.AppendEmpty())
+	if es.Len() > 0 {
+		mf.exemplars[gk] = es
+	}
+}
+
+func addExemplarToSlice(pe exemplar.Exemplar, e pmetric.Exemplar) {
+	e.SetTimestamp(pcommon.NewTimestampFromTime(time.UnixMilli(pe.Ts)))
+	e.SetDoubleVal(pe.Value)
+	for _, lb := range pe.Labels {
+		switch lb.Name {
+		case traceIDKey:
+			var tid [16]byte
+			b, _ := hex.DecodeString(lb.Value)
+			copyToLowerBytes(tid[:], b)
+			e.SetTraceID(tid)
+		case spanIDKey:
+			var sid [8]byte
+			b, _ := hex.DecodeString(lb.Value)
+			copyToLowerBytes(sid[:], b)
+			e.SetSpanID(sid)
+		default:
+			e.FilteredAttributes().EnsureCapacity(len(pe.Labels))
```

Comment:
This should be called before the for loop.

Reply:
Moved before for loop